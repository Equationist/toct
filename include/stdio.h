/* stdio.h - Standard Input/Output Library for CompilerKit C Compiler */

#ifndef _STDIO_H
#define _STDIO_H

/* Standard I/O types */
typedef struct {
    int _fd;
    int _flags;
    char* _buffer;
} FILE;

/* Standard streams */
extern FILE* stdin;
extern FILE* stdout; 
extern FILE* stderr;

/* Standard I/O functions */
int printf(const char* format, ...);
int fprintf(FILE* stream, const char* format, ...);
int sprintf(char* str, const char* format, ...);
int snprintf(char* str, int size, const char* format, ...);

int scanf(const char* format, ...);
int fscanf(FILE* stream, const char* format, ...);
int sscanf(const char* str, const char* format, ...);

int putchar(int c);
int puts(const char* s);
int fputs(const char* s, FILE* stream);
int fputc(int c, FILE* stream);

int getchar(void);
char* gets(char* s);
char* fgets(char* s, int size, FILE* stream);
int fgetc(FILE* stream);
int ungetc(int c, FILE* stream);

FILE* fopen(const char* filename, const char* mode);
FILE* freopen(const char* filename, const char* mode, FILE* stream);
int fclose(FILE* stream);
int fflush(FILE* stream);

long ftell(FILE* stream);
int fseek(FILE* stream, long offset, int whence);
void rewind(FILE* stream);
int feof(FILE* stream);
int ferror(FILE* stream);
void clearerr(FILE* stream);

/* File position constants */
#define SEEK_SET 0
#define SEEK_CUR 1
#define SEEK_END 2

/* End of file constant */
#define EOF (-1)

/* Buffer size */
#define BUFSIZ 8192

/* Null pointer */
#ifndef NULL
#define NULL ((void*)0)
#endif

#endif /* _STDIO_H */