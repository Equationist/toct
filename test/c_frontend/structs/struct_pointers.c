/* Test struct pointers and arrow operator */
#include <stdio.h>

struct Person {
    char name[20];
    int age;
};

void print_person(struct Person *p) {
    printf("Name: %s, Age: %d\n", p->name, p->age);
}

int main() {
    struct Person person1 = {"Alice", 25};
    struct Person person2 = {"Bob", 30};
    struct Person *ptr = &person1;
    
    printf("Direct access: ");
    printf("Name: %s, Age: %d\n", person1.name, person1.age);
    
    printf("Pointer access: ");
    printf("Name: %s, Age: %d\n", ptr->name, ptr->age);
    
    printf("Via function: ");
    print_person(&person2);
    
    ptr = &person2;
    ptr->age = 31;
    printf("After ptr->age = 31: ");
    print_person(ptr);
    
    return 0;
}