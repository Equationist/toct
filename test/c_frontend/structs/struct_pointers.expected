Direct access: Name: Alice, Age: 25
Pointer access: Name: Alice, Age: 25
Via function: Name: Bob, Age: 30
After ptr->age = 31: Name: Bob, Age: 31